### PR TITLE
Fireworks after christmas and during january

### PR DESCRIPTION
**Pull request by mads256h**
Screenshot (ignore the bats):

### CODE REVIEW

**Comment by Mast3rwaf1z** [comment]
new feature works on my machine :+1:

**Review by Mast3rwaf1z — APPROVED** [review]

**Comment by mads256h** [comment]
why draft? :(

**Comment by krestenlaust** [comment]
By my understanding, merging this feature would render the Stregsystem unusable at new year's

**Comment by mads256h** [comment]
No look at the screenshot i have changed the text color to white

**Comment by krestenlaust** [comment]
I'm blind. I'll do some code review some time today, then we'll get it merged!

File: stregsystem/static/stregsystem/fireworks.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        }
+
+        canvas.used = true;
+        explosions.push(explosion);
+
+        return;
+      }
+    }
+
+    console.log("COULD NOT FIND UNUSED CANVAS");
```
Comment (krestenlaust) [inline stregsystem/static/stregsystem/fireworks.js:242]: Interesting state to suddenly be in

File: stregsystem/static/stregsystem/fireworks.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+    const deltaTime = (now - rocketCanvas.lastTime) / 1000;
+    rocketCanvas.lastTime = now;
+
+    if (now >= nextRocketTime) {
+      nextRocketTime = now + rocketConfig.timeBetweenRockets * 1000 + Math.random() * rocketConfig.timeRandomness * 1000;
+
+
+      const endX = Math.random() * (width - 300) + 150;
+      const endY = Math.random() * (height - 300) + 150;
```
[the remark this answers is not in the document]
Comment (mads256h) [reply stregsystem/static/stregsystem/fireworks.js]: Do we really not expect that? xD
Fixed

File: stregsystem/static/stregsystem/fireworks.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+          gravityFactor: 0
+        };
+
+
+        const sparkleColor = generateColor();
+        const sparkleColor2 = generateColor();
+
+        const useSparkleColorForFlash = Math.random() > 0.5;
+
+        for (let i = 0; i < 160; i++) {
```
[the remark this answers is not in the document]
Comment (mads256h) [reply stregsystem/static/stregsystem/fireworks.js]: Fixed

File: stregsystem/static/stregsystem/fireworks.js (hunk 1 above)
Comment (mads256h) [reply stregsystem/static/stregsystem/fireworks.js]: I made it explain itself
Its not a critical problem if we hit it

**Review by krestenlaust — APPROVED** [review]
LGTM

**Pull request merged**